packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -338,14 +342,22 @@ function registerCommands(
   );
 }
 
-function registerIsvAuthWatcher(): vscode.Disposable {
-  const isvAuthWatcher = vscode.workspace.createFileSystemWatcher(
-    path.join('.sfdx', 'sfdx-config.json')
-  );
-  isvAuthWatcher.onDidChange(uri => setupGlobalDefaultUserIsvAuth());
-  isvAuthWatcher.onDidCreate(uri => setupGlobalDefaultUserIsvAuth());
-  isvAuthWatcher.onDidDelete(uri => setupGlobalDefaultUserIsvAuth());
-  return vscode.Disposable.from(isvAuthWatcher);
+function registerIsvAuthWatcher(context: vscode.ExtensionContext) {
+  if (
+    vscode.workspace.workspaceFolders instanceof Array &&
+    vscode.workspace.workspaceFolders.length > 0
+  ) {
+    const configPath = path.join(
+      vscode.workspace.workspaceFolders[0].uri.fsPath,
```

Comment:
In my testing, I found that we need to pass the full workspace path to `vscode.workspace.createFileSystemWatcher`,rather than the relative path from the workspace dir to the config file